src/main/java/com/google/sps/utils/QuizTimingPropertiesUtils.java

```diff
@@ -67,28 +68,29 @@ public Long getTimestampProperty(String entity, DatastoreService datastore) {
             Entity fetched_item = pq.asList(FetchOptions.Builder.withLimit(1)).get(0);
             return (Long) fetched_item.getProperty("quiz_timestamp");
         } 
-        //log.severe("Zero Items Quered");
         log.log(Level.SEVERE, "No results for query {0}", entity);
         return null;
     }
  
     //This function checks if the user has taken the quiz yet by comparing their timestamp with the quiz's timestamp
-    public Boolean userTookQuiz(String usersQuizTime, String currentQuizTime) {
-        return (usersQuizTime.compareTo(currentQuizTime) > 0);
+    public Boolean userTookQuiz(Long usersQuizTime, Long currentQuizTime) {
+        try {
+            return usersQuizTime > currentQuizTime;
+        } catch(NullPointerException e ) {
+            log.log(Level.SEVERE, "Null result for parameter");
+            return null;
+        }  
     }
 
     //This function checks to see if the quiz is outdated
-     public Boolean isQuizOutdated(Long current_quiz_time) {
-        String quiz_date;
+     public Boolean isQuizOutdated(Long current_quiz_time) {    
+        Long today_date = System.currentTimeMillis();
         try {
-            quiz_date = DateFormat.getDateInstance().format(current_quiz_time);
-        } catch(IllegalArgumentException e) {
+            return today_date > current_quiz_time;
```

Comment:
What is current_quiz_time? I thought it was the time the quiz was created, which would be in the past. If so, this will always return `true`.

`System.currentTimeMillis() - current_quiz_time` would give the age of the quiz in milliseconds

Reply:
it is the time the quiz is created but everyday it gets set to a new time. So if today's date is greater than the quiz time then we know we have to create a new quiz question and subsequently update the time of the quiz

Reply:
`System.currentTimeMillis()` is not today's date -- it's the number of milliseconds since Jan 1 1970 at midnight UTC.